pkg/graph/resolvers/plan_data_exchange_approach.go

```diff
@@ -27,53 +37,167 @@ func PlanDataExchangeApproachGetByModelPlanID(logger *zap.Logger, store *storage
+				isSettingToComplete, ok := isDataExchangeApproachComplete.(bool)
+				if !ok {
+					return nil, fmt.Errorf("unable to update plan data exchange approach, isDataExchangeApproachComplete is not a bool")
+				}
+
+				// Check if time has been set or is the default value
+				if existing.MarkedCompleteDts == nil && isSettingToComplete {
+					deaChangedToComplete = true
+					existing.MarkedCompleteBy = &principal.Account().ID
+					existing.MarkedCompleteDts = models.TimePointer(time.Now().UTC())
+					existing.Status = models.DataExchangeApproachStatusCompleted
+				} else if !isSettingToComplete {
+					existing.MarkedCompleteBy = nil
+					existing.MarkedCompleteDts = nil
+					existing.Status = models.DataExchangeApproachStatusInProgress
+				}
+
+				// Remove isDataExchangeApproachComplete from the changes map, since it's part of the
+				// input type but is NOT actually part of the model
+				delete(changes, "isDataExchangeApproachComplete")
+			}
+
+			// Update the base task list section
+			err = BaseStructPreUpdate(logger, existing, changes, principal, store, true, true)
+			if err != nil {
+				return nil, err
+			}
+
+			// Update the plan data exchange approach
+			retDataExchangeApproach, err := storage.PlanDataExchangeApproachUpdate(tx, logger, existing)
+
+			if deaChangedToComplete {
+				TrySendDataExchangeApproachNotifications(
+					ctx,
+					existing,
+					logger,
+					emailService,
+					emailTemplateService,
+					emailAddressBook,
+					principal,
+					tx,
+				)
+			}
+
+			return retDataExchangeApproach, err
+		},
+	)
+
 	if err != nil {
 		return nil, err
 	}
-	if existing.Status == models.DataExchangeApproachStatusReady {
-		existing.Status = models.DataExchangeApproachStatusInProgress
-	}
 
-	// Check if the 'changes' map contains the 'isDataExchangeApproachComplete' key and that the
-	// 'isDataExchangeApproachComplete' is different from the existing value
-	if isDataExchangeApproachComplete, ok := changes["isDataExchangeApproachComplete"]; ok {
-		isSettingToComplete, ok := isDataExchangeApproachComplete.(bool)
-		if !ok {
-			return nil, fmt.Errorf(" unable to update plan data exchange approach, isDataExchangeApproachComplete is not a bool")
-		}
+	return updatedDataExchangeApproach, nil
+}
 
-		// Check if time has been set or is the default value
-		if existing.MarkedCompleteDts == nil && isSettingToComplete {
-			// We do not actually store the isDataExchangeApproachComplete field in the database
-			// so we need to remove it from the changes map
-
-			existing.MarkedCompleteBy = &principal.Account().ID
-			existing.MarkedCompleteDts = models.TimePointer(time.Now().UTC())
-			existing.Status = models.DataExchangeApproachStatusCompleted
-		} else if !isSettingToComplete {
-			existing.MarkedCompleteBy = nil
-			existing.MarkedCompleteDts = nil
-			existing.Status = models.DataExchangeApproachStatusInProgress
-		}
+func TrySendDataExchangeApproachNotifications(
```

Comment:
You have two really similar functions that do this, is there a reason they both exist? Looks like the other one is only called in dbseed `resolvers.SendDataExchangeApproachMarkedCompleteNotification`. Could / Should this be unified?

Reply:
I went over it, specific functions are for SRP. These functions are verbose, following clean code principles. While they can be unified, I don't think this improves the code.